nengo_bones/templates/remote.sh.template

```diff
@@ -93,12 +93,36 @@ See https://docs.travis-ci.com/user/encrypting-files/ for more details.
         REMOTE_STATUS=0
         cd ~/"$BUILD_DIR"
         {% block remote_install %}
-        echo "$ ({{ host }}) Installing miniconda"
-        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh --quiet -O miniconda.sh
-        bash miniconda.sh -b -p ./miniconda
+        echo "$ ({{ host }}) Downloading miniconda"
+        DOWNLOAD_STATUS=0
+        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh --quiet -O miniconda.sh || DOWNLOAD_STATUS=1
+        if [ "\$DOWNLOAD_STATUS" -gt 0 ]; then
+            echo "({{ host }}) Failed to download miniconda. Exiting.";
+            exit \$DOWNLOAD_STATUS;
+        fi
+        for i in \$(seq 1 10); do
+            echo "$ ({{ host }}) Installing miniconda (try \$i)"
+            INSTALL_STATUS=0
+            bash miniconda.sh -b -p ./miniconda || INSTALL_STATUS=1
```

Comment:
If the `./miniconda` directory already exists (e.g., because we're re-running a previous build) then this will fail, which we don't want to treat as a critical error. We could either delete the `./miniconda` directory ahead of time if it exists, or we could make this whole download/install block conditional on the `./miniconda` directory not existing. I'd probably lean towards just deleting it if it exists, so that we're starting from a clean slate each time.

Reply:
I do delete it if it fails, see below.

Reply:
Yes but if the `.miniconda` installation succeeds, but then a unit test fails or something like that so you restart the build, then the `.miniconda` directory will still be there.

Reply:
Ahh, I see what you mean. That sounds like a good idea.

Reply:
I guess after it failed the first time (because the `.miniconda` directory exists), then it would delete it and then try again, so that kind of solves the problem 🤷.

Reply:
Ok, I've added a fixup.